### ISSUE TITLE
Helper methods

### ISSUE THREAD

**Issue by jamesplease** [jmeas/redux-simple-resource#98]
Did a request fail? Are some resources loading? These are common questions, and RSR lets you answer pretty much any question you would want to about XHRs. But it is quite verbose to do so! We should add helper methods.

I had three ideas:

1. Boolean helper methods. i.e.;

```js
isReadPending(state, ['books.23', 'requests']);
// true if Book with ID 23 is loaded, and if a bulk read of requests loaded
```

2. xhrStatus helper methods, i.e.;

```js
getXhrStatus(state, 'books.20.relationships.author.23', 'updating')
// returns the xhrStatus for the update action of this book
```

3. changed statuses. Did a value change from one thing to another?

```js
didReadFail(currentResources, nextResources, 'books.20', 'delete');
// => true if the delete failed, false otherwise
```

**Comment by jamesplease** [comment]
The idea of a "resource path" is pretty nice, but there are issues with the idea of a label (see #99 for more on what a label is). I don't want a DSL that's any weirder than object/keys, so something like `resourceName{label}` isn't an API that I'm interested in.

One idea would be:

```js
any(
  status: 'PENDING',
  state: { movies, books },
  statuses: [
    getReadStatus('movies', {view: 'pasta'}),
    getReadStatus('books.24')
  ]
])
```

then there would be an analogous `all` for checking for success.

but, yikes that just seems rather complex ☹️

**Comment by jamesplease** [comment]
Alright, some updates...

Paths are great, but since they map naturally to objects, I'll support both. In other words, both of these point to the same location:

```js
'books.23'

{
  resourceName: 'books',
  id: 23
}
```

This supports labels/views/anything else I want naturally:

```js
{
  resourceName: 'books',
  whatever: 'sandwiches'
}
```

there just won't be an equivalent string syntax for these extra things, which I'm alright with.

---

I'm thinking of just having one helper. It gives you meta for one resource, like:

```js
getStatus(state, 'read', 'books.23')
// => {pending: false, failed: true, succeeded: false}
```

and can also aggregate:

```js
getStatus(state, 'read', ['books.23', 'friends'])
```

The reason for this is that you usually care to aggregate the same action (and honestly, it is just reads). If you wanted to aggregate different statuses across different actions, you can do that using good ol JS like so:

```js
[
  getStatus(state, 'delete', 'books.23').pending,
  getStatus(state, 'update', 'books.24').failed
].every(v => v);

// or...

[
  getStatus(state, 'delete', 'books.23').pending,
  getStatus(state, 'update', 'books.24').failed
].some(v => v);
```

---

For checking for "events" (moments when it goes from pending to not), you could do:

```js
const wasPending = getStatus(prevState, 'read', ['books.23']).pending;
const isFailed = getStatus(nextState, 'read', ['books.23']).failed;
const didFail = wasPending && isFailed;
```

which doesn't seem so bad to me.

**Comment by jamesplease** [comment]
Resolved in #113 ; will be released in v2.0
